### DISCUSSION TITLE
How to implement multiple submit buttons per form

### DISCUSSION THREAD

**Discussion by robkuz** [unpoly/unpoly#529] [Q&A]
Hi,

I have a huge form with multiple "subforms" and tables.
In the tables for example I have buttons that add or delete rows.
I want those buttons to trigger a (partial) update on the table - aka submit the formdata to the server.
However each of those buttons should target a different route on the server (e.g. "add_row_table_A", "delete_row_table_B" etc.).

How can I do this?

thx
Robertj

**Comment by adam12** [comment]
I would probably use the `[formaction]` HTML attribute on the [`<button>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button).

**Reply by robkuz** [reply]
yeah ... kinda. using formaction does not intercept and rerender via unpoly but does a navigation. sigh. I am writing a macro (or a I try).
Thanks thou for the hint

**Reply by adam12** [reply]
Is Unpoly configured on the `<form>` element? Also, just confirming you're not nesting `<form>` elements (they don't like that).

**Reply by robkuz** [reply]
yes it is. and no nested forms.

**Reply by adam12** [reply]
Strange. Perhaps you can enable logging and see if there is anything interesting in the logs. `up.log.enable()`.

If you can share the markup of the form, it might help as well.
